Automattic/wp-calypso
# A4A Dev Sites: Adding loading state to the Launch site button in Site Settings

Fixes https://github.com/Automattic/dotcom-forge/issues/9217

## Proposed Changes

* Set the "Launch site" button to busy state when the launch request is in progress. 
* Disable the "Refer to client" button when the launch request is in progress.

#### After PR Applied
Notice the state of the "Launch site" and "Refer to client" buttons.

##### Successful Launch Request
https://github.com/user-attachments/assets/58f527e1-e2d0-4d7d-b3dc-d3556ad692c0

##### Erroronous Launch Request
https://github.com/user-attachments/assets/1168a728-3a3f-4929-aaac-1a4492f896bf

## Why are these changes being made?


* To enhance UX.

## Testing Instructions



* Go to http://calypso.localhost:3000/settings/general/your-dev-site.wpcomstaging.com
* Click on the "Launch site" button.
* Confirm the action in the popup.
* Optional: At this stage, you may want to open up your browser dev tools, and in the Networks tab adjust the throtling to 3G to slow down the launch request.
* Ensure the "Launch site" button renders as busy, and that the "Refer to client" button becomes disabled. 

## Pre-merge Checklist
